pkgs/applications/editors/emacs/elisp-packages/manual-packages/lspce/package.nix

```
@@ -1,34 +1,116 @@
 {
   lib,
-  callPackage,
   f,
   markdown-mode,
   melpaBuild,
   nix-update-script,
   yasnippet,
+  fetchFromGitHub,
+  rustPlatform,
 }:
 
 let
-  lspce-module = callPackage ./module.nix { };
+  src = fetchFromGitHub {
+    owner = "zbelial";
+    repo = "lspce";
+    rev = "fd320476df89cfd5d10f1b70303c891d3b1e3c81";
+    hash = "sha256-KnERYq/CvJhJIdQkpH/m82t9KFMapPl+CyZkYyujslU=";
+  };
 in
 melpaBuild {
   pname = "lspce";
-  inherit (lspce-module) version src meta;
+  version = "1.1.0-unstable-2024-07-14";
+
+  inherit src;
 
   packageRequires = [
     f
     markdown-mode
     yasnippet
   ];
 
-  # to compile lspce.el, it needs lspce-module.so
-  files = ''(:defaults "${lib.getLib lspce-module}/lib/lspce-module.*")'';
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    hash = "sha256-BMbkdWsVSXRNt8kqOs05376cGnGnivHGvEugX0p3bVc=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.cargoSetupHook
+    rustPlatform.cargoBuildHook
+    rustPlatform.cargoInstallHook
+    rustPlatform.cargoCheckHook
+  ];
+
+  cargoBuildType = "release";
+
+  # Copy the default buildPhase here and switch cd to pushd/popd in
+  # order to not cause rebuild of other Emacs lisp packages.
+  buildPhase = ''
+    runHook preBuild
+
+    pushd "$NIX_BUILD_TOP"
+
+    emacs --batch -Q \
+        -L "$NIX_BUILD_TOP/package-build" \
+        -l "$melpa2nix" \
+        -f melpa2nix-build-package \
+        $ename $melpaVersion $commit
+
+    popd
+
+    runHook postBuild
+  '';
+
+  # Copy the default installPhase here because adding the dynamic
+  # module to the package tarball needs to be done at a specific time,
+  # i.e., between cargoInstallHook and elpa2nix-install-package.
+  # It cannot be done in preInstall because preInstall, as an implicit
+  # string hook, runs before cargoInstallHook, as a hook in
+  # preInstallHooks.
+  installPhase = ''
+    runHook preInstall
+
+    archive="$NIX_BUILD_TOP/packages/$ename-$melpaVersion.el"
+    if [ ! -f "$archive" ]; then
+        archive="$NIX_BUILD_TOP/packages/$ename-$melpaVersion.tar"
+    fi
+
+    echo "add the dynamic module to the package tarball because it is needed for compilation"
```

```suggestion
    # add the dynamic module to the package tarball because it is needed for compilation
```

I can't see a reason to echo this info by default. 
If there is some reason to do this, it should use some logging functionality instead.

---

I slightly prefer logging this info because it is a bit "unusual".

> logging functionality

You mean things like `nixInfoLog`, right?